diff --git a/configure.ac b/configure.ac
index ed45aaa..b5fa113 100644
--- a/configure.ac
+++ b/configure.ac
@@ -90,8 +90,8 @@ PKG_CHECK_EXISTS([xorg-server >= 1.16],
 AC_ARG_WITH(xorg-conf-dir,
             AS_HELP_STRING([--with-xorg-conf-dir=DIR],
                            [Default xorg.conf.d directory [[default=$prefix/share/X11/xorg.conf.d/]]]),
-            [xorgconfdir="$withval"],
-            [xorgconfdir="$prefix/share/X11/xorg.conf.d"])
+            [configdir="$withval"],
+            [configdir="$prefix/share/X11/xorg.conf.d"])
 AC_SUBST(configdir)
 AM_CONDITIONAL(HAS_XORG_CONF_DIR, [test "x$sysconfigdir" != "x"])
 
